libs/cardano-ledger-core/src/Cardano/Ledger/Core/TxCert.hs

```diff
@@ -87,6 +89,26 @@ class
   -- | Extract staking credential from any certificate that can unregister such credential
   lookupUnRegStakeTxCert :: TxCert era -> Maybe (Credential 'Staking (EraCrypto era))
 
+  -- | Compute the total deposits from a list of certificates.
+  getTotalDepositsTxCerts ::
+    Foldable f =>
+    PParams era ->
+    -- | Check whether stake pool is registered or not
+    (KeyHash 'StakePool (EraCrypto era) -> Bool) ->
+    f (TxCert era) ->
+    Coin
+
+  -- | Compute the total refunds from a list of certificates.
+  getTotalRefundsTxCerts ::
+    Foldable f =>
+    PParams era ->
+    -- | Lookup current deposit for Staking credential if one is registered
+    (Credential 'Staking (EraCrypto era) -> Maybe Coin) ->
+    -- | Lookup current deposit for DRep credential if one is registered
+    (Credential 'DRepRole (EraCrypto era) -> Maybe Coin) ->
+    f (TxCert era) ->
+    Coin
```

Comment:
Just a thought (maybe not a good one). But have we considered having a default implementation for these. After all, this is the same for all era's before Conway.

Reply:
I certainly did give it a thought on multiple occasions

Default implementation would be good if it was the same for all future eras, rather than the first few. Because we wouldn't need to worry about specifying it for new instances. Providing a default implementation that works only for the first few eras could lead to a dangerous state where we forget to supply the correct one in the future era. 

Although I totally understand the desire to do that, since it would save a few lines of code. I am guilty of doing just that on multiple occasions, due to my laziness.